lib/auth/grpcserver_test.go

```diff
@@ -2214,3 +2218,191 @@ func TestCustomRateLimiting(t *testing.T) {
 		})
 	}
 }
+
+type mockAuthorizer struct {
+	ctx *Context
+	err error
+}
+
+func (a mockAuthorizer) Authorize(context.Context) (*Context, error) {
+	return a.ctx, a.err
+}
+
+type mockTraceClient struct {
+	err   error
+	spans []*tracepb.ResourceSpans
+}
+
+func (m mockTraceClient) Start(ctx context.Context) error {
+	return nil
+}
+
+func (m mockTraceClient) Stop(ctx context.Context) error {
+	return nil
+}
+
+func (m *mockTraceClient) UploadTraces(ctx context.Context, protoSpans []*tracepb.ResourceSpans) error {
+	m.spans = protoSpans
+
+	return m.err
+}
+
+func TestExport(t *testing.T) {
+	t.Parallel()
+	uploadErr := trace.AccessDenied("failed to upload")
+
+	const user = "user"
+
+	validateTaggedSpans := func(forwardedFor string) require.ValueAssertionFunc {
+		return func(t require.TestingT, i interface{}, i2 ...interface{}) {
+			require.NotEmpty(t, i)
+			resourceSpans, ok := i.([]*tracepb.ResourceSpans)
+			require.True(t, ok)
+
+			for _, resourceSpan := range resourceSpans {
+				for _, scopeSpan := range resourceSpan.ScopeSpans {
+					var foundForwardedTag bool
+					for _, span := range scopeSpan.Spans {
+						for _, attribute := range span.Attributes {
+							if attribute.Key == forwardedTag {
+								foundForwardedTag = true
+								require.Equal(t, forwardedFor, attribute.Value.GetStringValue())
+								break
+							}
+						}
+						require.True(t, foundForwardedTag)
+					}
+				}
+			}
+		}
+	}
+
+	testSpans := []*tracepb.ResourceSpans{
+		{
```

Comment:
style:
```suggestion
	testSpans := []*tracepb.ResourceSpans{{
```
for all the slices with a single item in them